cuda/extensions.bzl

```diff
@@ -2,7 +2,23 @@
 
 load("//cuda/private:repositories.bzl", "local_cuda")
 
+cuda_toolkit = tag_class(attrs = {
+    "name": attr.string(doc = "Name for the toolchain repository"),
```

Comment:
This brings up an interesting and not fully explored aspect of Bzlmod. I *think* that having a `name` attribute on a tag class that ends up as a repository name verbatim should be considered an antipattern and avoided if possible: If names collide between two transitive dependencies that both use rules_cuda, users can't really do anything about their broken build.

I described this issue and a possible workaround in https://github.com/bazelbuild/rules_go/pull/3443. In the particular situation of rules_cuda, a simpler solution could be to ignore tags of modules other than the root module (`module_ctx.is_root`). That simplifies the implementation for now and allows you to explore this design space in more detail in the future.

CC @Wyverald

Reply:
FYI this pattern came from @alexeagle 's PR to rules-template: https://github.com/bazel-contrib/rules-template/pull/46/files